Zend/zend_execute.c

```diff
@@ -1722,16 +1723,126 @@ ZEND_API ZEND_COLD void zend_wrong_string_offset_error(void)
 	zend_throw_error(NULL, "%s", msg);
 }
 
+ZEND_COLD static zend_result ZEND_FASTCALL get_deprecation_suffix_from_attribute(HashTable *attributes, zend_class_entry* scope, zend_string **message_suffix)
+{
+	*message_suffix = zend_empty_string;
+
+	if (!attributes) {
+		return SUCCESS;
+	}
+
+	zend_attribute *deprecated = zend_get_attribute_str(attributes, "deprecated", sizeof("deprecated")-1);
+
+	if (!deprecated) {
```

Comment:
Should this assert instead? Same above.

Reply:
No, this is required for compatibility: If an extension manually specifies the `ZEND_ACC_DEPRECATED` bit, a function will be deprecated, but not have the attribute: https://wiki.php.net/rfc/deprecated_attribute#to_existing_extensions

> The attribute will not be automatically applied to existing functions having the doc comment, but extension authors are encouraged to apply the attribute for consistency reasons.

Reply:
I see. I did not consider the old deprecated flag for internal functions, without attributes.